integrations/access/servicenow/client.go

```diff
@@ -0,0 +1,312 @@
+	DateTimeFormat = "2006-01-02 15:04:05"
+)
+
+var (
+	incidentBodyTemplate = template.Must(template.New("incident body").Parse(
+		`{{.User}} requested permissions for roles {{range $index, $element := .Roles}}{{if $index}}, {{end}}{{ . }}{{end}} on Teleport at {{.Created.Format .TimeFormat}}.
+{{if .RequestReason}}Reason: {{.RequestReason}}{{end}}
+{{if .RequestLink}}To approve or deny the request, proceed to {{.RequestLink}}{{end}}
+`,
+	))
+	reviewNoteTemplate = template.Must(template.New("review note").Parse(
+		`{{.Author}} reviewed the request at {{.Created.Format .TimeFormat}}.
+Resolution: {{.ProposedState}}.
+{{if .Reason}}Reason: {{.Reason}}.{{end}}`,
+	))
+	resolutionNoteTemplate = template.Must(template.New("resolution note").Parse(
+		`Access request has been {{.Resolution}}
+{{if .ResolveReason}}Reason: {{.ResolveReason}}{{end}}`,
+	))
+)
+
+// Client is a wrapper around resty.Client.
+type Client struct {
+	ClientConfig
+
+	client *resty.Client
+}
+
+// ClientConfig is the config for the servicenow client.
+type ClientConfig struct {
+	// APIEndpoint is the endpoint for the Servicenow API
+	// api url of the form  https://instance.service-now.com/ with optional trailing '/'
+	APIEndpoint string
+
+	// WebProxyURL is the Teleport address used when building the bodies of the incidents
+	// allowing links to the access requests to be built
+	WebProxyURL *url.URL
+	// ClusterName is the name of the Teleport cluster
+	ClusterName string
+
+	// Username is the username used by the client for basic auth.
+	Username string
+	// APIToken is the token used for basic auth.
+	APIToken string
+}
+
+// NewClient creates a new Servicenow client for managing incidents.
+func NewClient(conf ClientConfig) (*Client, error) {
+	if err := conf.checkAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	client := resty.NewWithClient(defaults.Config().HTTPClient)
+	client.SetBaseURL(conf.APIEndpoint).
+		SetHeader("Content-Type", "application/json").
+		SetHeader("Accept", "application/json").
+		SetBasicAuth(conf.Username, conf.APIToken)
+	return &Client{
+		client:       client,
+		ClientConfig: conf,
+	}, nil
+}
+
+func (conf ClientConfig) checkAndSetDefaults() error {
+	if conf.APIEndpoint == "" {
+		return trace.BadParameter("missing required field: APIEndpoint")
+	}
+	return nil
+}
+
+func errWrapper(statusCode int, body string) error {
+	switch statusCode {
+	case http.StatusForbidden:
+		return trace.AccessDenied("servicenow API access denied: status code %v: %q", statusCode, body)
+	case http.StatusRequestTimeout:
+		return trace.ConnectionProblem(nil, "connecting to servicenow API :status code %v: %q", statusCode, body)
+	}
+	return trace.Errorf("connecting to servicenow API status code %v: %q", statusCode, body)
+}
+
+// CreateIncident creates an servicenow incident.
+func (snc Client) CreateIncident(ctx context.Context, reqID string, reqData RequestData) (Incident, error) {
```

Comment:
Should we define methods on the pointer type instead?
https://github.com/golang/go/wiki/CodeReviewComments#receiver-type
> If the receiver is a struct, array or slice and any of its elements is a pointer to something that might be mutating, prefer a pointer receiver, as it will make the intention clearer to the reader.

```suggestion
func (snc *Client) CreateIncident(ctx context.Context, reqID string, reqData RequestData) (Incident, error) {
```